File: modelo/alumno.cs
using System.Data;
using System;
public class alumno : oObjeto
{
    public string Nombre { get; set; }
    public string Apellido { get; set; }
    public string Numerodoc { get; set; }
    public int Legajo { get; set; }
    public string Condicion { get; set; }
    public string Correo { get; set; }
    public DateTime Fechanac { get; set; }
    public int Iddomicilio { get; set; } 
    public int Idusuario { get; set; }
    public alumno() { }
    public alumno(DataRow dr)
    {
        this.Id = Convert.ToInt32(dr["id"]);
        this.Nombre = dr["nombre"].ToString();
        this.Apellido = dr["apellido"].ToString();
        this.Numerodoc = dr["numerodoc"].ToString();
        this.Condicion = dr["condicion"].ToString();
        this.Correo = dr["correo"].ToString();
        this.Fechanac = Convert.ToDateTime(dr["fechanac"]);
        this.Iddomicilio = Convert.ToInt32(dr["iddomicilio"]); 
        this.Idusuario = Convert.ToInt32(dr["idusuario"]);
        this.Legajo = Convert.ToInt32(dr["legajo"]);
    }

}

File: ClientApp/src/app/clases/alumno.ts
export class alumno{
    id : number;
    nombre : string;
    apellido : string;
    numerodoc : string;
    condicion : string;
    correo : string;
    fechanac : Date; 
    iddomicilio : number;
    idusuario : number;
    
    constructor(id: string, nombre:string, apellido:string, numerodoc:string, condicion: string, correo:string, fechanac:string, iddomicilio :string, idusuario:string){
        //el orden de os parametros afecta la vista de los input en el modal
        this.id = +id;
        this.nombre = nombre;
        this.apellido =  apellido;
        this.numerodoc = numerodoc;
        this.condicion = condicion;
        this.correo = correo ;
        this.fechanac = new Date(fechanac); 
        this.iddomicilio = +iddomicilio;
        this.idusuario = +idusuario;
         // formateo a number el string // parseInt(val

    }
    
}